File: Frontend/types/post.ts
import { postRequestSchema } from "@/schemas/postSchema";
import { UUID } from "@/types";
import { z } from "zod";

export type PostListItemResponse = {
  id: UUID;
  title: string;
  slug: string;
  thumbnailUrl: string;
  createdDate: Date;
};

export type PostResponse = {
  id: UUID;
  title: string;
  content: string;
  imageUrl: string;
  isPublished: boolean;
  tags: string[];
  createdDate: Date;
  updatedDate: Date;
};

export type PostUpdateResponse = {
  id: UUID;
  title: string;
  slug: string;
  content: string;
  imageUrl: string;
  isPublished: boolean;
  tags: string[];
};

export type PostRequest = z.infer<typeof postRequestSchema>;

File: Backend/Domain/Entities/Post.cs
namespace Domain.Entities;

public class Post : BaseEntity
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public string Slug { get; set; } = null!;
    public string Content { get; set; } = null!;
    public string ThumbnailUrl { get; set; } = null!;
    public string ImageUrl { get; set; } = null!;
    public bool IsPublished { get; set; }
    public List<string> Tags { get; set; } = [];
}
